include/nbl/asset/IRayTracingPipeline.h

```diff
@@ -0,0 +1,171 @@
+#ifndef _NBL_ASSET_I_RAY_TRACING_PIPELINE_H_INCLUDED_
+#define _NBL_ASSET_I_RAY_TRACING_PIPELINE_H_INCLUDED_
+
+#include "nbl/asset/IShader.h"
+#include "nbl/asset/IPipeline.h"
+
+#include <span>
+
+
+namespace nbl::asset
+{
+
+class IRayTracingPipelineBase : public virtual core::IReferenceCounted
+{
+  public:
+    struct SShaderGroupsParams
+    {
+      constexpr static inline uint32_t ShaderUnused = 0xffFFffFFu;
+
+      struct SGeneralShaderGroup
+      {
+        uint32_t shaderIndex = ShaderUnused;
+      };
+
+      struct SHitShaderGroup
+      {
+        uint32_t closestHitShaderIndex = ShaderUnused;
+        uint32_t anyHitShaderIndex = ShaderUnused;
+        uint32_t intersectionShaderIndex = ShaderUnused;
+      };
+
+      SGeneralShaderGroup raygenGroup;
+      std::span<SGeneralShaderGroup> missGroups;
+      std::span<SHitShaderGroup> hitGroups;
+      std::span<SGeneralShaderGroup> callableGroups;
```

Comment:
I'd clean-up/simplify the naming of this even more
```cpp
struct SIndex
{
    constexpr static inline uint32_t Unused = 0xffFFffFFu;

    uint32_t index = Unused;
};
struct SHitGroup
{
   SIndex closestHit = SIndex::Unused;
   SIndex anyHit = SIndex::Unused;
   SIndex intersection = SIndex::Unused;
};

SIndex raygen;
std::span<SIndex> misses;
std::span<SHitGroup>
std::span<SIndex> callables;
```

the reason is that they're all inside `SShaderGroupParams`, so don't want to write `..shader..::...shader....shader` (a tautology)